### PR TITLE
Better template doc

### LINKED ISSUE #232: Better documentation for the templates
### Generic Issue

* xscen version: 0.6.15

### Description
I've been helping @vindelico get accustomed to `xscen` in the last few weeks, and he had a few suggestions for our templates (especially the first 1, which is the most complete one).

1. Be as exhaustive as possible.
In some places in the config, we only write the arguments that are being overwritten. The reason why some arguments are missing wasn't clear. Something like this would be preferable:
```
  climatological_mean: # automatically passed to the function
    window: 30
    min_periods:  # Default value, so you would not need to include it in a real configuration file.
    interval: 10
    periods: [['1951', '2100']]
    to_level: climatology
```

2. Better explain why certain functions (i.e. `compute_deltas`) are under specific categories (i.e `aggregate`)
```
# Functions under this section are associated to xscen/aggregate.py
aggregate:
  dask:
    n_workers: 4
    threads_per_worker: 4
    memory_limit: "6GB"
  ...
```

3. Better order/indicate which function gets sent automatically to `xscen` through `parse_config` and which doesn't.
```
aggregate:
  # These entries do not correspond to an `xscen` function, but if you plan to have multiple variations of them
  # (i.e. to fine-tune dask workers depending on the task to perform), then categorizing them under `aggregate` 
  # makes it easier to locate them within the CONFIG object when you want to perform tasks related to the
  # `aggregate` module, such as when computing `climatological_mean`
  dask:
    n_workers: 4
    threads_per_worker: 4
    memory_limit: "6GB"
  input:
    clim:
      processing_level: indicators
    delta:
      processing_level: climatology
  save:
    mode: 'o'

  # These entries exactly match `xscen` functions located under the `aggregate` module, so the parameters 
  # specified below will automatically override default values. 
  # See https://xscen.readthedocs.io/en/latest/notebooks/6_config.html for more details on how it works.
  climatological_mean: 
    window: 30
    interval: 10
    periods: [['1951', '2100']]
    to_level: climatology
  compute_deltas: 
    kind: "+"
    reference_horizon: "1991-2020"
    to_level: 'delta'

```

### PR DESCRIPTION
**Pull request by juliettelavoie**
### Pull Request Checklist:
- [x] This PR addresses an already opened issue (for bug fixes / features)
    - This PR fixes #232
- [ ] (If applicable) Documentation has been added / updated (for bug fixes / features).
- [ ] (If applicable) Tests have been added.
- [x] This PR does not seem to break the templates.
- [x] HISTORY.rst has been updated (with summary of main changes).
  - [x] Link to issue (:issue:`number`) and pull request (:pull:`number`) has been added.

### What kind of change does this PR introduce?

* improve template doc

### Other information:
Let's do it here instead of #233

### CODE REVIEW

**Comment by RondeauG** [comment]
I see a few things that could be adjusted too. I'll do it directly in the branch.

File: templates/1-basic_workflow_with_config/workflow1.py [hunk 1]
```diff
@@ -23,86 +23,107 @@
...
+            # Arguments are not passed automatically, because the config is different for each type of data.
+            # Therefore, we must manually send the 'type_dict' entry to the search_data_catalogs function.
             cat = xs.search_data_catalogs(**type_dict["search_data_catalogs"])
 
-            # iterate over ids from the search
+            # Iterate over the datasets that matched the search
+            # 'ds_id' is the ID of the dataset, 'dc' is the sub-catalog for this dataset
             for ds_id, dc in cat.items():
-                # attrs of current iteration that are relevant now
+                # These are some relevant attributes that will be added to the dataset after extraction.
```
Comment (juliettelavoie) [inline templates/1-basic_workflow_with_config/workflow1.py:72]:
```suggestion
                # These are some relevant attributes that are used to check if the task was done already and to write the output path.
```

File: templates/1-basic_workflow_with_config/properties1.yml [hunk 2]
```diff
@@ -1,18 +1,23 @@
...
-      long_name: 2th quantile of the minimum temperature
+      long_name: 2nd quantile of the minimum temperature
     input:
-      da: tasmin
+      da: tasmin  # Since properties are generic, the input name must be specified.
     parameters:
-      q: 0.98
+      q: 0.02
       group: time.season
   maximum_length_of_warm_spell:
```
Comment (vindelico) [inline templates/1-basic_workflow_with_config/properties1.yml:16]: This part of the example (maximum_length_of_warm_spell) didn't work for me, but that's a different issue.

**Review by vindelico — APPROVED** [review]
These comments make the workflow and the underlying workings much more transparent, thanks!

File: templates/1-basic_workflow_with_config/workflow1.py [hunk 3]
```diff
@@ -23,87 +23,109 @@
...
+    # 7) update the ProjectCatalog
+
     # --- REGRID ---
     if "regrid" in CONFIG["tasks"]:
-        # get input and iter over datasets
-        input_dict = pcat.search(**CONFIG["regrid"]["inputs"]).to_dataset_dict(**tdd)
+        # Search the ProjectCatalog for the results of the previous step, then iterate over each dataset.
+        # We usually don't have to rely on search_data_catalogs anymore after the initial extraction, because the content of the ProjectCatalog is smaller and more manageable.
+        # In most cases, we can just use the search function with the 'type' and 'processing_level' attributes.
+        input_dict = pcat.search(**CONFIG["regrid"]["inputs"]).(**tdd)
```
Comment (RondeauG) [inline templates/1-basic_workflow_with_config/workflow1.py:128]:
```suggestion
        input_dict = pcat.search(**CONFIG["regrid"]["inputs"]).to_dataset_dict(**tdd)
```

Comment (RondeauG) [reply templates/1-basic_workflow_with_config/workflow1.py:128]: @vindelico Those are actually needed!

File: templates/1-basic_workflow_with_config/workflow1.py [hunk 4]
```diff
@@ -113,30 +135,31 @@
...
 
     # --- BIAS ADJUST ---
     if "biasadjust" in CONFIG["tasks"]:
-        # iter over each variable that needs to be adjusted
+        # Bias adjustment differs for each variable, so we need to iterate over them.
         for var, ba_dict in CONFIG["biasadjust"].items():
-            # get all input simulations and iter over them
-            dict_sim = pcat.search(**ba_dict["sim_inputs"]).to_dataset_dict(**tdd)
+            # Search the ProjectCatalog for the results of the previous step, then iterate over each dataset.
+            dict_sim = pcat.search(**ba_dict["sim_inputs"]).(**tdd)
```
Comment (RondeauG) [inline templates/1-basic_workflow_with_config/workflow1.py:162]:
```suggestion
            dict_sim = pcat.search(**ba_dict["sim_inputs"]).to_dataset_dict(**tdd)
```

File: templates/1-basic_workflow_with_config/workflow1.py [hunk 5]
```diff
@@ -189,28 +217,28 @@
...
                         path = f"{CONFIG['paths']['task']}".format(**cur)
                         xs.save_to_zarr(ds_clean, path, **CONFIG["cleanup"]["save"])
                         pcat.update_from_ds(ds=ds_clean, path=path)
 
     # --- RECHUNK and store final daily data ---
     if "rechunk" in CONFIG["tasks"]:
-        # get inputs and iter over them
-        dict_input = pcat.search(**CONFIG["rechunk"]["inputs"]).to_dataset_dict(**tdd)
+        # Search the ProjectCatalog for the results of the previous step, then iterate over each dataset.
+        dict_input = pcat.search(**CONFIG["rechunk"]["inputs"]).(**tdd)
```
Comment (RondeauG) [inline templates/1-basic_workflow_with_config/workflow1.py:241]:
```suggestion
        dict_input = pcat.search(**CONFIG["rechunk"]["inputs"]).to_dataset_dict(**tdd)
```

File: templates/1-basic_workflow_with_config/workflow1.py [hunk 6]
```diff
@@ -243,10 +271,10 @@
 
     # --- DIAGNOSTICS ---
     if "diagnostics" in CONFIG["tasks"]:
-        # compute properties (and measures) on different kinds of data (ref, sim,scen)
+        # The properties and measures that we want to compute are different for each type of data (ref, sim, scen), so we need to iterate over them.
         for kind, kind_dict in CONFIG["diagnostics"]["kind"].items():
-            # iterate on inputs
-            dict_input = pcat.search(**kind_dict["inputs"]).to_dataset_dict(**tdd)
+            # Search for the right datasets and iterate over them
+            dict_input = pcat.search(**kind_dict["inputs"]).(**tdd)
```
Comment (RondeauG) [inline templates/1-basic_workflow_with_config/workflow1.py:277]:
```suggestion
            dict_input = pcat.search(**kind_dict["inputs"]).to_dataset_dict(**tdd)
```

File: templates/1-basic_workflow_with_config/properties1.yml (hunk 2 above)
Comment (juliettelavoie) [reply templates/1-basic_workflow_with_config/properties1.yml:16]: can you make an issue to explain the problem you had, please ? then, we can try to fix it!

File: templates/1-basic_workflow_with_config/workflow1.py (hunk 3 above)
Comment (vindelico) [reply templates/1-basic_workflow_with_config/workflow1.py]: Oups! I was searching for these and I guess I replaced them. Obviously, they are needed :-)

**Pull request merged**